source/includes/inkling-reference/_schema.html.md

```diff
@@ -112,98 +120,141 @@ end
 
 ###### Primitive Types
 
-Inkling has a set of primitive types which are used in schema declarations. The integer suffix indicates the size in bits of the type.
+Click the syntax tab to show the Inkling set of primitive types which are used in schema declarations. The
+integer suffix indicates the size in bits of the type. Integer types beginning
+with 'U' are unsigned. 
 
-> primitiveType :=
 
-```c
-Double | Float64 | Float32 | Int8 | Int16 | Int32 |
-Int64 | UInt8 | UInt16 | UInt32  | UInt64 | Bool | String
+```plaintext
+primitiveType ::=
+  Double | Float64 | Float32 | Int8 | Int16 | Int32 |
+  Int64 | UInt8 | UInt16 | UInt32  | UInt64 | Bool | String
 ```
 
 ###### Structured Types
 
-Inkling supports the types Matrix and Luminance (more to come).
+Inkling currently supports the types Matrix and Luminance. This list will be
+expanded.
 
-> structureDcln      :=
+See the schema declaration syntax for the complete syntax of declaration.
 
-```c
-( Luminance | Matrix )
-  structureInit
-  <name>
+```plaintext
+structure_type ::= 
+  Luminance | Matrix
 ```
+‍
+###### Constrained Types
 
+Constrained types are supported in schemas and also in [lessons][1]. They are constrained by means of a special type of expression called a range expression.
 
-> structureInit      :=
+```inkling
+  # Example: Range expression
+
+  schema Schema1
+    Int8 {0:3:10} field   # start:step:stop
+  end
+ 
+ # Curly braces delineate the range expression.
+ # The values in this range are: (0, 3, 6, 9).
+ # Note that the specification of 10 as the stop does not mean
+ # 10 must be in the range (because it is a limit, not 
+ # necessarily an endpoint).
+```
 
-```c
-'('
+A range expression has the effect of constraining the values of the type to values defined by the range expression. In a schema this constrains the values in the field. In lessons this constrains the values of the placeholder being configured. In both cases the syntax is the same.
+
+Select the Inkling tab to see an example of a constrained type in a schema definition. 
+
+### Constrained Type Syntax
 
-luminanceInit | matrixInit
+Select the Syntax tab to view the constrained type syntax.
 
-')'
+```plaintext
+constrainedType ::=
+primitiveType
+'{'
+  start ':' [ step':']? stop      # A 'colon range'. Specifies 'step' 
+  |
+  start '.' '.' stop ':' numSteps # A 'dot range'. Specifies 'numsteps'.
+'}'
 ```
 
-> luminanceInit      :=
+‍
+
+###### Range Expression Rules
 
-```c
-integerLiteral  ',' integerLiteral
+There are three forms of range expressions which Inkling supports. 
+Select the Inkling tab to see an examples of each type.
 
-matrixInit         :=
- '(' concreteType
-  [ ',' concretetype ]*
- ')'
+* **Value list range expression**
 
-',' integerLiteral [ ',' integerLiteral]*
+```inkling
+# Example: Value list range expression
+
```

Comment:
I was commenting on how this one has an extra line break in between the comment and the Inkling code. Just wanted them all to be the same, line or no.

<img width="521" alt="screen shot 2017-03-17 at 5 08 58 pm" src="https://cloud.githubusercontent.com/assets/5640420/24066988/732bb736-0b34-11e7-82c0-cc982e48d984.png">